### PR TITLE
Make enter select largest folder if nothing is selected

### LINKED ISSUE #37: auto select single folder
When you enter a folder and there's only one subfolder, it makes sense to automatically select it.
Maybe also consider auto selecting the biggest folder if there are many subfolders.
Because currently it takes an extra button press to select anything.

### PR DESCRIPTION
**Pull request by redzic**
Fixes #37
Although, the way it is now, it will only _select_ (but not `cd` into) the largest folder if nothing is selected. From the wording "... the app goes into the largest folder" in the comment of #37 I wasn't really sure if it should `cd` into the largest folder or just select it. However, if should also `cd` into it, it should be a very trivial change in the `handle_enter` function.

### CODE REVIEW

**Comment by redzic** [comment]
It looks like the tests are failing because this PR calls slightly different operations to draw to the terminal
However, it seems like the feature is working as intended when I run it locally, so maybe the tests should be changed?

**Review by imsnif — CHANGES_REQUESTED** [review]
Hey @redzic, happy you picked this up! Sorry for not being clearer in my wording. I actually didn't think of your solution of pressing enter to select. I tried out both versions (you're right, it really is a small change to have it enter the selected folder) and IMO it's better if it also enters the folder (pressing ENTER when nothing is selected enters the largest folder). But maybe that's just me... what do you think?
Otherwise, I've left some small nitpicks in the PR.

**Comment by redzic** [comment]
Yeah tbh, I think it's probably better if it also enters the directory because I tried using the feature and if you keep doing it multiple times it's a little inconvenient because you have to press enter twice each time. And I think I understand why the tests are failing now, thanks!

I'll make the suggested changes, thank you!

File: src/state/tiles/board.rs
Comment (redzic) [reply src/state/tiles/board.rs:67]: lol I guess I didn't even realize that the tiles were already sorted by size. Well that makes things much more efficient!

With the `use` statement, it looks like another thing called `Folder` (`state::files::Folder`) is already imported in `board.rs`, so I'll just import `FileType` and in the closure for the `filter` method just write `FileType::Folder`

**Comment by imsnif** [comment]
Hey, only one more thing I forgot! Could you add a test for this? I think you can copy/paste one of the "move_X_and_enter" tests and remove the movement. What say you?

**Comment by redzic** [comment]
Sure, I'll try writing a test for it! Sounds not too bad if I can just slightly copy and modify already existing tests lol

**Comment by imsnif** [comment]
Great! I'm copy-pasting something I wrote in another thread about our snapshot tests. I hope to have this in a CONTRIBUTING.md file ASAP, but for now so you won't be stuck:

About snapshot testing with insta: we use insta (https://github.com/mitsuhiko/insta) for snapshot testing. It gives us the assert_snapshot macro. The first time it is run, it has no snapshot to compare to, so it creates one in the snapshots folder. If you install cargo-insta on your computer, you will be able to review them by doing `cargo insta review` after the snapshot is created. You'll see the new snapshot (in this case, the "are you sure?" modal) and be able to approve it or reject it. Once you approve it, the test should pass when you run it again.

**Comment by redzic** [comment]
Oops, kind of have a lot more commits than necessary because I was a little confused with the bottom `asssert!`s. Hopefully the test should look something like this? Thank you for the help btw!

**Comment by imsnif** [comment]
This is a really cool feature, I'm having a lot of fun with it in long directory chains :)

The test looks great and everything else does as well.

**Review by imsnif — APPROVED** [review]

**Comment by imsnif** [comment]
Btw, no worries about the commits. I'm squashing them anyway.

**Pull request merged**
